tests/unit/geo.tcl

```diff
@@ -360,6 +368,14 @@ start_server {tags {"geo"}} {
         assert {[lindex $res 0] eq "Catania"}
     }
 
+    test {GEOSEARCH the box spans -180° or 180°} {
+        r del points
+        r geoadd points 179.5 36 point1
+        r geoadd points -179.5 36 point2
+        assert_equal {point1 point2} [r geosearch points fromlonlat 179 37 bybox 400 400 km asc]
+        assert_equal {point2 point1} [r geosearch points fromlonlat -179 37 bybox 400 400 km asc]
+    }
+
     test {GEOADD + GEORANGE randomized test} {
         set attempt 30
```

Comment:
i think it may be nice to re-introduce some of the changes from the previous PR here too:
```
    foreach {type} {byradius bybox} {
    test "GEOSEARCH fuzzy test - $type" {
        if {$::accurate} { set attempt 300 } else { set attempt 30 }
```
even if we're using the same algorithm as the C code to to decide if the point is inside the box or not (I agree we don't want the `pointInPolygon` and our odd polygon shape).
but still running this fuzz test on a box in addition to a radius may be able to smoke out some bug.

Reply:
i agree, I will re-introduce foreach bybox test and use same algorithm `pointInRectangle` in tcl.